src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Godot;
+
+/// <summary>
+///   Tweaked color picker defines some custom ColorPicker functions.
+/// </summary>
+public class TweakedColourPicker : ColorPicker
+{
+    private readonly CheckButton hsvCheckButton;
+
+    private readonly CheckButton rawCheckButton;
+
+    private readonly Button addPresetButton;
+
+    private bool hsvModeDisabled;
+
+    private bool rawModeDisabled;
+
+    public TweakedColourPicker()
+    {
+        var baseNode = GetChild(4).GetChild(4);
+        hsvCheckButton = baseNode.GetChild<CheckButton>(0);
+        hsvCheckButton.Connect("toggled", this, nameof(OnHSVButtonToggled));
+        rawCheckButton = baseNode.GetChild<CheckButton>(1);
+        rawCheckButton.Connect("toggled", this, nameof(OnRawButtonToggled));
+
+        addPresetButton = GetChild(7).GetChild<Button>(0);
+        addPresetButton.Connect("mouse_entered", this, nameof(OnMouseEnteredAddPresetButton));
+    }
+
+    [Export]
+    public bool HSVModeDisabled
+    {
+        get => hsvModeDisabled;
+        set
+        {
+            hsvModeDisabled = value;
+            UpdateControl();
+        }
+    }
+
+    [Export]
+    public bool RawModeDisabled
+    {
+        get => rawCheckButton.Disabled;
+        set
+        {
+            rawModeDisabled = value;
+            UpdateControl();
+        }
+    }
+
+    public new bool HsvMode
+    {
+        get => base.HsvMode;
+        set
+        {
+            if (hsvModeDisabled)
+                return;
+
+            base.HsvMode = value;
+        }
+    }
+
+    public new bool RawMode
+    {
+        get => base.RawMode;
+        set
+        {
+            if (rawModeDisabled)
+                return;
+
+            base.RawMode = value;
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+        UpdateControl();
+    }
+
+    public void OnMouseEnteredAddPresetButton()
+    {
+        addPresetButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_ADD_PRESET");
+    }
+
+    private void OnHSVButtonToggled(bool isOn)
+    {
+        if (!isOn && rawModeDisabled)
+            rawCheckButton.Disabled = true;
+    }
+
+    private void OnRawButtonToggled(bool isOn)
+    {
+        if (!isOn && hsvModeDisabled)
+            hsvCheckButton.Disabled = true;
+    }
+
+    private void UpdateControl()
```

Comment:
`UpdateControl` was an example name, would be clearer if this is instead was named `UpdateToggleButtons`.